### PR TITLE
👷(ci) add docker publication github actions

### PR DESCRIPTION
**Pull request by wilbrdt**
## Proposal

Automate release of Docker images.

## Proposal

- Add two new workflows to build/publish APP and API docker images
- Rename other Github Actions files

Note: For now, we don't require the `service` workflow (that build/test/lint code) to be successful before triggering the `release` workflow. This would need some refactoring (using callable workflows?)

### CODE REVIEW

File: .github/workflows/release-app.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+          # Display either:
+          # - DOCKER_TAG: main (Git branch main)
+          # or
+          # - DOCKER_TAG: 1.1.2 (Git tag v1.1.2)
+          echo "DOCKER_TAG: ${DOCKER_TAG} (Git $GITHUB_REF_TYPE $GITHUB_REF_NAME)"
+          docker tag warren-tdbp:$GITHUB_SHA apuiavignonuniversity/warren-tdbp:app-$DOCKER_TAG
+          # Tag as "latest" for stable release git tags
+          if [[ "$GITHUB_REF_TYPE" == "tag" ]]; then
+              docker tag warren-tdbp:$GITHUB_SHA apuiavignonuniversity/warren-tdbp:app-latest
+          fi
```
Comment (quitterie-lcs) [inline .github/workflows/release-app.yml]:
```suggestion
          [[ "$GITHUB_REF_TYPE" == "tag" ]] && docker tag warren-tdbp:$GITHUB_SHA apuiavignonuniversity/warren-tdbp:app-latest
```

File: .github/workflows/release-app.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+          DOCKER_TAG=$([[ "$GITHUB_REF_TYPE" == "branch" ]] && echo $GITHUB_REF_NAME || echo $GITHUB_REF_NAME | sed 's/^v//')
+          # Display either:
+          # - DOCKER_TAG: main (Git branch main)
+          # or
+          # - DOCKER_TAG: 1.1.2 (Git tag v1.1.2)
+          echo "DOCKER_TAG: $DOCKER_TAG (Git $GITHUB_REF_TYPE $GITHUB_REF_NAME)"
+          docker push fundocker/warren:app-$DOCKER_TAG
+          if [[ "$GITHUB_REF_TYPE" == "tag" ]]; then
+            docker push fundocker/warren:app-latest
+          fi
```
Comment (quitterie-lcs) [inline .github/workflows/release-app.yml]: Same suggestion here

**Review by quitterie-lcs — COMMENTED** [review]
I propose to remove the usage of if condition and replace it with logical AND

File: .github/workflows/release-app.yml (hunk 1 above)
Comment (wilbrdt) [reply .github/workflows/release-app.yml]: I agree with the one-liner, but the line is quite long already :sweat_smile:
I think I'll leave it as is, if you don't mind!

File: .github/workflows/release-app.yml (hunk 2 above)
Comment (wilbrdt) [reply .github/workflows/release-app.yml]: Leaving it as is, so we keep the same general format

File: src/frontend/Dockerfile [hunk 3]
```diff
@@ -42,3 +42,9 @@ RUN apt-get update && \
 # Un-privileged user running the application
 USER ${DOCKER_USER:-1000}
 
+
+# -- Production --
+FROM core as production
+
+# Un-privileged user running the application
+USER ${DOCKER_USER:-1000}
```
Comment (jmaupetit) [inline src/frontend/Dockerfile]: Not sure this is required as this image is used for development / build system, and will not run in production.

Comment (wilbrdt) [reply src/frontend/Dockerfile]: You're right! Removed it

**Review by jmaupetit — APPROVED** [review]

**Pull request merged**